moduleframework/dockerlinter.py

```diff
@@ -161,11 +162,62 @@ def get_specific_label(self, label_name=None):
         if label_name is None:
             return []
         label_list = self.get_docker_labels()
-        return [label_name in label_list]
+        return [label_list[key] for key in label_list.keys() if label_name == key]
 
-    def check_baseruntime(self):
+    def check_from_is_first(self):
         """
-        Function returns docker labels
-        :return: label dictionary
+        Function checks if FROM directive is really first directive.
+        :return: True if FROM is first, False if FROM is not first directive
+        """
+        if self.dfp_structure[0].get('instruction') == 'FROM':
+            return True
+        else:
+            return False
+
+    def check_from_directive_is_valid(self):
         """
-        return [x for x in self.docker_dict.get(FROM,[]) if "baseruntime/baseruntime" in x]
+        Function checks if FROM directive contains valid format like is specified here
+        http://docs.projectatomic.io/container-best-practices/#_line_rule_section
+        Regular expression is: ^[a-z0-9.]+(\/[a-z0-9\D.]+)+$
+        Example registry:
+            registry.fedoraproject.org/f26/etcd
+            registry.fedoraproject.org/f26/flannel
+            registry.access.redhat.com/rhscl/nginx-18-rhel7
+            registry.access.redhat.com/rhel7/rhel-tools
+            registry.access.redhat.com/rhscl/postgresql-95-rhel7
+
+        :return:
+        """
+        correct_format = False
+        struct = self.dfp_structure[0]
+        if struct.get(INSTRUCT) == 'FROM':
+            p = re.compile("^[a-z0-9.]+(\/[a-z0-9\D.]+)+$")
+            if p.search(struct.get('value')) is not None:
+                correct_format = True
+        return correct_format
+
+    def check_two_dnf_run_sections(self):
+        value = 0
+        for struct in self.dfp_structure:
+            if struct.get(INSTRUCT) == RUN:
+                value += 1
+        if int(value) > 1:
+            return False
+        return True
+
+    def check_user_number(self):
```

Comment:
can be also string, see full spec: https://docs.docker.com/engine/reference/builder/#user

Reply:
Thanks I did not find it before implementation.
That means, container maintainer should know what he is writing in Dockerfile. We have no possibility how to check this.
I will remove it.